services/motion/builtin/state/state_test.go

```diff
@@ -717,6 +734,455 @@ func TestState(t *testing.T) {
 		test.That(t, err, test.ShouldBeNil)
 		test.That(t, ps8, test.ShouldBeEmpty)
 	})
+
+	t.Run("ttl", func(t *testing.T) {
+		t.Parallel()
+		pollTillDuration := time.Millisecond * 500
+		sReq := state.Request{
+			TTL:              time.Millisecond * 100,
+			TTLCheckInterval: time.Millisecond * 10,
+			Logger:           logger,
+		}
+		sleepCheckDuration := sReq.TTLCheckInterval * 2
+		sleepTTLDuration := sReq.TTLCheckInterval * 2
+		s, err := state.NewState(sReq)
+		test.That(t, err, test.ShouldBeNil)
+		defer s.Stop()
+
+		// no plan statuses as no executions have been created
+		ps, err := s.ListPlanStatuses(motion.ListPlanStatusesReq{})
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, ps, test.ShouldBeEmpty)
+
+		preExecution := time.Now()
+
+		req := motion.MoveOnGlobeReq{ComponentName: myBase}
+
+		// start execution, then stop it to bring it to terminal state
+		executionID1, err := state.StartExecution(ctx, s, req.ComponentName, req, executionWaitingForCtxCancelledPlanConstructor)
+		test.That(t, err, test.ShouldBeNil)
+
+		// stop execution, to show that it still shows up within the ttl
+		err = s.StopExecutionByResource(myBase)
+		test.That(t, err, test.ShouldBeNil)
+
+		// start execution, leave it running
+		executionID2, err := state.StartExecution(ctx, s, req.ComponentName, req, executionWaitingForCtxCancelledPlanConstructor)
+		test.That(t, err, test.ShouldBeNil)
+
+		// wait till check interval past
+		time.Sleep(sleepCheckDuration)
+
+		listPlanStatuses := func(waitTillLen int) func() (struct {
+			ps  []motion.PlanStatusWithID
+			err error
+		}, bool,
+		) {
+			return func() (struct {
+				ps  []motion.PlanStatusWithID
+				err error
+			}, bool,
+			) {
+				st := struct {
+					ps  []motion.PlanStatusWithID
+					err error
+				}{}
+				ps, err := s.ListPlanStatuses(motion.ListPlanStatusesReq{})
+				if err == nil && len(ps) == waitTillLen {
+					st.ps = ps
+					st.err = err
+					return st, true
+				}
+				return st, false
+			}
+		}
+
+		// TODO: Come back here
```

Comment:
TODO:

- [ ] Test outputs of GetPlan for above code
- [ ] Test behavior of replanning before & after TTL
- [ ] Test behavior of failure before & after TTL
- [ ] Test behavior of success before & after TTL